initfiles/bash/etc/systemd/system/thorslaves-launch.sh.in

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+while getopts ":c:" opts; do
+    case "${opts}" in
+        c)
+            instanceName="${OPTARG}"
+            ;;
+        *)
+            help
+            ;;
+    esac
+done
+shift $((OPTIND - 1))
+action=$1
+
+exec 2>/tmp/thorslaves-launch.debug
+set -x
+
+[[ -z "${instanceName}" || -z "${action}" ]] && exit 1
+
+cwd=$(pwd)
+if [[ "${cwd}" != "@RUNTIME_PATH@/${instanceName}" ]]; then
+    [[ ! -d "@RUNTIME_PATH@/${instanceName}" ]] && exit 1
+    cd @RUNTIME_PATH@/${instanceName}
+fi
+
+source ${instanceName}.cfg
+
+slaveIps=($(@EXEC_PATH@/daliadmin server=$DALISERVER clusternodes ${THORNAME} slaves timeout=2 1>/dev/null 2>&1; uniq slaves))
+if [[ -z $slaveIps ]]; then
+	slaveIps=($(uniq slaves 2>&1))
+fi
+[[ -z $slaveIps ]] && exit 1
+
+numOfNodes=${#slaveIps[@]}
+
+for ((i=0;i<${#slaveIps[@]};i++)); do
+    for ((c=0;c<${slavespernode};c++)); do
+        __slavePort=$((${THORSLAVEPORT} + (${c} * ${localthorportinc}))) 
+        __slaveNum=$((${i} + 1 + (${c} * ${#slaveIps[@]})))
+        ssh -o LogLevel=QUIET -o StrictHostKeyChecking=no -o BatchMode=yes -i ${SSHidentityfile} ${SSHusername}@${slaveIps[$i]} "/bin/bash -c '@ADMIN_PATH@/thorslaves-exec.sh ${action} thorslave_${instanceName}_${__slaveNum} ${__slavePort} ${__slaveNum} ${THORNAME} ${THORMASTER} ${THORMASTERPORT}'"
```

Comment:
Serially calling ssh vs previous frunssh is going to be painful (slow).

I don't see "localthor" check here.